nChannels not according to paper

PabloRR100 opened the pull request:
Hi @bamos,

According to the paper, the number of channels is twice the grow_rate only for the BC type DenseNets.
Since you are giving room to not have bottleneck, we should check that before giving nChannels. If not BC, channels = 16.

I also included one doubt on the forward method and a number of parameters sanity check that does not match with the BC reported by the authors.

Regards,
Pablo

PabloRR100 committed:
nChannels not according to paper

Hi @bamos,

According to the paper, the number of channels is twice the grow_rate only for the BC type DenseNets.
Since you are giving room to not have bottleneck, we should check that before giving nChannels. If not BC, channels = 16.

I also included one doubt on the forward method and a number of parameters sanity check that does not match with the BC reported by the authors.

Regards,
Pablo

PabloRR100 committed:
DenseNets-BC parameters fixed

Solved the problems with the parameters on the DenseNets-BC

bamos commented on densenet.py:
```diff
@@ -111,6 +112,49 @@ def forward(self, x):
+        #out = torch.squeeze(F.avg_pool2d(F.relu(self.bn1(out)), 8)) ## Why did you relu and bn again?
+        out = self.avgpool(8)
+        # out = F.log_softmax(self.fc(out))
+        out = self.fc(out)                      # You can define the softmax within the loss function right?
```
I don't think so, this part shouldn't have been changed

bamos commented on densenet.py:
```diff
@@ -111,6 +112,49 @@ def forward(self, x):
-        out = torch.squeeze(F.avg_pool2d(F.relu(self.bn1(out)), 8))
-        out = F.log_softmax(self.fc(out))
+        #out = torch.squeeze(F.avg_pool2d(F.relu(self.bn1(out)), 8)) ## Why did you relu and bn again?
+        out = self.avgpool(8)
```
If I understand correctly, adding `self.avgpool` is a stylistic change. I prefer keeping the `torch.functional` verison. Also shouldn't `out` be input the pooling layer? How is this code working?

bamos reviewed (changes requested):
Thanks! I added a few comments inline to the PR.

PabloRR100 committed:
Missed out in the pooling

My point is that CNN models normally outputs the output of the fully connected layer.
Then this vector is passed to the Criterion(preds,targets) with the correct labels to compute the loss (CrossEntropy).

And yes the self.avgpool and the flatten are only to keep the style constant and more readable but the functionality is the same

PabloRR100 committed:
Minor change
